### DISCUSSION TITLE
What do you use to write code?

### DISCUSSION THREAD

**Discussion by Steve0Greatness** [theforumhelpers/theforumhelpers.github.io#283] [General]
What I mean is what program(Exs: Replit, Code Sandbox, Glitch, Github's Editor) do you use to develop this site.
Personally, I use Replit.

**Comment by Accio1** [comment]
Different developers might use different editors. I use Atom.

**Reply by BarelySmooth** [reply]
Just curious, why do you prefer Atom over VSCode?

**Reply by Accio1** [reply]
@BarelySmooth I guess I am just more used to Atom. I never really tried VSCode.

**Comment by leahcimto** [comment]
I use VSCode. I used to use Atom, but switched to VSCode and I like VSCode better.

**Reply by Steve0Greatness** [reply]
you can import Github repos into VSC?

**Reply by leahcimto** [reply]
I just clone it. You could use VSCode's github system I think. I haven't tried that though.

**Reply by leahcimto** [reply]
I clone with Github Desktop though since I'm not good with the command line.

**Reply by Steve0Greatness** [reply]
> I clone with Github Desktop though since I'm not good with the command line.

understandable, have a great day.

**Reply by Accio1** [reply]
> I clone with Github Desktop though since I'm not good with the command line.

+1 Git is too confusing

**Reply by FunctionalMetatable** [reply]
IMO if you know force pushing you know the way to go in git
`git push --force`

**Reply by imdevarsh** [reply]
Git CLI is awesome.

**Reply by BarelySmooth** [reply]
> Git CLI is awesome.

If you don't really *need* CLI, there are lots of GUI options that are far better than the command line version.

**Reply by imdevarsh** [reply]
I can't push a tag from GH Desktop.

**Reply by BarelySmooth** [reply]
Sure, they have limitations. But GUIs do the job 99% of the time.

**Comment by aidan-crockett** [comment]
I use Github Desktop and that makes it really easy to clone and edit repos along with VSCode which I like too.
(Github's '-bad-' editor uses [Codemirror](http://codemirror.net))

**Reply by Steve0Greatness** [reply]
As far as I can tell, the online editor doesn't use that.
or it doesn't use many of it's features if it does.

**Reply by aidan-crockett** [reply]
![Screen Shot 2021-08-03 at 3 23 56 PM](https://user-images.githubusercontent.com/82768218/128081221-9bc93c6a-5eef-4177-9b8f-ab6946408639.png)

**Reply by Steve0Greatness** [reply]
@gosoccerboy5, oh.

**Reply by aidan-crockett** [reply]
> or it doesn't use many of it's features if it does.

Probably.

**Comment by BarelySmooth** [comment]
I love VS Code and its integration with GitHub (and GitHub Desktop)

I also sometimes use the VCS Integration directly from within VSCode. I still use GitHub Desktop for most things - it just has a really good UI.
![image](https://user-images.githubusercontent.com/83728060/128132819-b5ae8a28-d733-44fa-9748-d3ed63c2180f.png)

What I really like about VS Code is the great extension marketplace. Even Microsoft makes some really useful extensions (like [Live Preview](https://github.com/microsoft/vscode-livepreview) – in the above image). That is also probably why I have three extensions for the same thing 🤔

I also like VS Code for [their YouTube Channel](https://youtu.be/hy_w_W4nug0)

**Reply by aidan-crockett** [reply]
wait wait wait - i've been using the [jeffalo.lived](//marketplace.visualstudio.com/items?itemName=jeffalo.lived) extension. is that the same as/similar to jeffalo.lived? (it seems to have a context menu option, which is [something I've wanted](https://github.com/jeffalo/lived/issues/5))

**Reply by BarelySmooth** [reply]
It is very similar. But I really like the feature which allows you to preview the website in VS Code itself without ever leaving it. It even has Devtools (mostly because VSCode is Electron based)

**Reply by aidan-crockett** [reply]
oh.

**Comment by Shluffy** [comment]
I use chrome app text (
![image](https://user-images.githubusercontent.com/81823039/128437244-53939d95-7549-4ba6-9ebd-4c4866d9738a.png)
), replit, or the github editor. I don't have any better options

**Reply by Accio1** [reply]
Chrome Text App is the MVP. I used it for ~2 years before switching to Atom and I still prefer the way it syntax highlights opposed to Atom.

**Reply by Shluffy** [reply]
MVCE (Most Valuable Code Editor)

**Comment by leahcimto** [comment]
I do use the github editor for small changes (like fixing a typo) though. It's just faster.

**Reply by Steve0Greatness** [reply]
makes sense.

**Reply by aidan-crockett** [reply]
You never know when even a tiny change may break something 😅

**Reply by leahcimto** [reply]
That happens way too often for me lol

**Reply by Shluffy** [reply]
I was using JSON and I formatted it like this:
```
[
{
"pokemon": "gible",
"amount": "1",
"event": "Community Day Gible, June 2021"
},
]
```

what is wrong here that broke everything. this is whole file

**Reply by aidan-crockett** [reply]
trailing comma (line 6)

**Reply by aidan-crockett** [reply]
[happened to me before](https://github.com/gosoccerboy5/gosoccerboy5.github.io/commit/d7156104)

**Comment by FunctionalMetatable** [comment]
i used to use atom but it was confusing and then i moved to vscode because jeffalo said i should so i did and i dont regret that advice vscode is really good with the builtin shell

**Reply by aidan-crockett** [reply]
The builtin shell is totally epic.

**Reply by aidan-crockett** [reply]
I don't know where I'd be without it

**Comment by BarelySmooth** [comment]
For quick changes, you could consider [The Dot](https://www.youtube.com/watch?v=ywUZOOzLX3c).

**Reply by FunctionalMetatable** [reply]
I never knew we could use something like codespaces

**Reply by aidan-crockett** [reply]
AAAAA!

**Reply by aidan-crockett** [reply]
that is so epic, I thought codespaces were closed off!!

you have made my day so much better barelysmooth
(also, Atom is Github's 'official' editor, I wonder why they stuck with VSC)

**Reply by BarelySmooth** [reply]
> that is so epic, I thought codespaces were closed off!!

Codespaces is still a thing (and its still only available as a paid option). The only difference between "the dot" and Codespaces is that Codespaces runs on GitHub's servers but "the dot" runs on the browser. This means features like terminal won't be present. They explained it [here](https://www.youtube.com/watch?v=ywUZOOzLX3c&lc=UgxTm3NU_qCv-bK-2n54AaABAg.9QvKK36FsO-9QwfabLOUne) (see highlighted reply).

**Reply by aidan-crockett** [reply]
oh :(
still extremely epic

**Reply by BarelySmooth** [reply]
Yes 😛 I hope they atleast add an option to preview html pages. That would really be convenient.

**Reply by aidan-crockett** [reply]
It could really reduce my reliance on offline editors if I could do that!

**Reply by Accio1** [reply]
> For quick changes, you could consider [The Dot](https://www.youtube.com/watch?v=ywUZOOzLX3c).

Wow! That is neat!

**Reply by aidan-crockett** [reply]
> > For quick changes, you could consider [The Dot](https://www.youtube.com/watch?v=ywUZOOzLX3c).
> 
> Wow! That is neat!

you'd have to get used to vscode to use it 😛

**Reply by Steve0Greatness** [reply]
Epic. But I'll still use Replit for bigger changes.

**Reply by leahcimto** [reply]
wow

**Reply by leahcimto** [reply]
I just tried it, it is so cool!

**Reply by SheepTester** [reply]
> (also, Atom is Github's 'official' editor, I wonder why they stuck with VSC)

Atom is quite old (old enough to have originally been mostly written in CoffeeScript, so back in the ES5 days). Especially after Facebook ditched atom-ide-ui and Microsoft acquired GitHub, it seems GitHub has been directing its attention towards the newer and cooler VS Code.

Also, Atom wasn't really made to work in the web browser. Electron was created after Atom, so it's probable that Atom heavily relies on Electron-specific APIs. Meanwhile, VSCode was probably designed to be somewhat browser-compatible; the text editor itself, [Monaco](https://microsoft.github.io/monaco-editor/), can be used like CodeMirror or Ace on web pages

**Reply by aidan-crockett** [reply]
Huh. Well, too bad for atom I guess
